lib/github-service.js

```diff
@@ -29,6 +29,44 @@ class GithubService {
     });
   }
 
+  async getAssociatedPullRequestNumberFromSha(){
+    const res = await this.getOctokitInstance().request('GET /repos/{owner}/{repo}/commits/{commit_sha}/pulls', {
+      owner: this.options.repo.split('/')[0],
+      repo: this.options.repo.split('/')[1],
+      commit_sha: this.options.refCommit,
+      body: 'Comment created by API',
+      headers: {
+        'X-GitHub-Api-Version': '2022-11-28'
+      }
+    });
+
+    // if no pull request associated with commit return null
+    if (res.data.length == 0){
+      return null;
+    } else {
+      return res;
+    }
+  }
+
+  async createPullRequestComment(report){
```

Comment:
Use one single method : decoratePullRequest
and use return Promise.allSettled(function1, function2)